app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -378,6 +379,8 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
 
             DefaultBrowserNotificationWorker.setDefaultBrowserNotificationIfNeeded(applicationContext)
         }
+
+        components.core.store.dispatch(SearchAction.RefreshSearchEnginesAction)
```

Comment:
It seems like the update behavior is covered between the `InitAction` during app startup or the dispatch from `DefaultLocaleSettingsController`. Do we also need this dispatch, or can we rely on the other two?

Reply:
This call covers the case when the app is running in the background and the user changes the language from system settings then comes back in the app. Removing this call will lead to the following behavior:

https://user-images.githubusercontent.com/32488956/203505119-032e6e66-11a4-4841-bfb3-c7b358094f21.mp4

Nevertheless you are right that it is not clear enough why the dispatch is there, therefore I will add a comment before, linking to this thread. Thanks for pointing it out!